src/LineChart/components/LineChartContainer.ts

```diff
@@ -0,0 +1,236 @@
+        try {
+            return style.split(";").reduce<{[key: string]: string}>((styleObject, line) => {
+                const pair = line.split(":");
+                if (pair.length === 2) {
+                    const name = pair[0].trim().replace(/(-.)/g, match => match[1].toUpperCase());
+                    styleObject[name] = pair[1].trim();
+                }
+                return styleObject;
+            }, {});
+        } catch (error) {
+            // tslint:disable-next-line no-console
+            console.log("Failed to parse style", style, error);
+        }
+
+        return {};
+    }
+
+    public static validateProps(props: LineChartContainerProps): string {
+        let errorMessage = "";
+        if (props.sourceType === "microflow" && !props.dataSourceMicroflow) {
+            errorMessage += ` data source type is set to 'Microflow' but 'Source microflow' is missing \n`;
+        }
+
+        return errorMessage && `Configuration error :\n\n ${errorMessage}`;
+    }
+
+    private resetSubscriptions(mxObject?: mendix.lib.MxObject) {
+        this.subscriptionHandles.forEach(mx.data.unsubscribe);
+        this.subscriptionHandles = [];
+
+        if (mxObject) {
+            this.subscriptionHandles.push(mx.data.subscribe({
+                callback: this.handleSubscription,
+                guid: mxObject.getGuid()
+            }));
+        }
+    }
+
+    private handleSubscription() {
+        this.fetchData(this.props.mxObject);
+    }
+
+    private fetchData(mxObject?: mendix.lib.MxObject) {
+        const { seriesEntity } = this.props;
+        if (mxObject && this.props.seriesEntity) {
+            if (this.props.sourceType === "xpath") {
+                const constraint = this.props.entityConstraint
+                    ? this.props.entityConstraint.replace("[%CurrentObject%]", mxObject.getGuid())
+                    : "";
+                const entityName = seriesEntity.indexOf("/") > -1
+                    ? seriesEntity.split("/")[seriesEntity.split("/").length - 1]
+                    : seriesEntity;
+                const XPath = "//" + entityName + constraint;
+                this.fetchByXPath(XPath);
+            } else if (this.props.sourceType === "microflow" && this.props.dataSourceMicroflow) {
+                this.fetchByMicroflow(mxObject.getGuid());
+            }
+
+        }
+    }
+
+    private fetchByXPath(xpath: string) {
+        window.mx.data.get({
+            callback: mxObjects => this.fetchDataFromSeries(mxObjects),
+            error: error => this.setState({
+                alertMessage: `An error occurred while retrieving data via XPath (${xpath}): ${error}`,
+                data: []
+            }),
+            xpath
+        });
+    }
+
+    private fetchByMicroflow(guid: string) {
+        const actionname = this.props.dataSourceMicroflow;
+        mx.ui.action(actionname, {
+            callback: mxObjects => {
```

Comment:
`callback: mxObjects => this.fetchDataFromSeries(mxObjects as mendix.lib.MxObject[]),`